test/testkit/mocks/backend.go

```diff
@@ -40,6 +40,10 @@ func (b *Backend) HTTPBackendConfigMap(name string) *LogBackendConfigMap {
 	return NewLogBackendConfigMap(name, b.namespace, b.exportedFilePath)
 }
 
+func (b *Backend) TLSBackendConfigMap(name, certPem, keyPem, caPem string) *TLSBackendConfigMap {
+	return NewTLSBackendConfigMap(name, b.namespace, b.exportedFilePath, certPem, keyPem, caPem, b.signalType)
+}
```

Comment:
Maybe let's instead pass the TLS option into the `Backend` constructor function and use the old `ConfigMap` function?